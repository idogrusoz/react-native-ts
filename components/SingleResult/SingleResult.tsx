import React, { FunctionComponent } from "react";
import { View, Text } from "../Themed";
import { Image, StyleSheet } from "react-native";
import { TouchableHighlight } from "react-native-gesture-handler";
import { useDispatch, useSelector } from "react-redux";
import { setLoading, setRecipe, setRecipeWithFetch } from "../../redux/actions/actions";
import { useNavigation } from "@react-navigation/native";
import { recipesSelector } from "../../redux/selectors";

type SingleResultProps = {
    id: number;
    title: string;
    image: string;
};

const SingleResult: FunctionComponent<SingleResultProps> = ({ id, title, image }) => {
    const { randomResults, isRandom } = useSelector(recipesSelector);
    const dispatch = useDispatch();
    const navigation = useNavigation();
    const handlePress = () => {
        dispatch(setLoading(true));
        if (isRandom) {
            dispatch(setRecipe(randomResults.filter((item) => item.id === id)[0]));
        } else {
            dispatch(setRecipeWithFetch(id));
        }
        navigation.navigate("Recipe");
    };
    return (
        <TouchableHighlight onPress={handlePress}>
            <View style={styles.wrapper}>
                <Image source={{ uri: image }} style={styles.image} />
                <Text style={styles.title} numberOfLines={1}>
                    {title}
                </Text>
            </View>
        </TouchableHighlight>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
    },
    image: {
        width: 156,
        height: 115,
    },
    title: {
        width: 115,
        textAlign: "center",
    },
});

export default SingleResult;
