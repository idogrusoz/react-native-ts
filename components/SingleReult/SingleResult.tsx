import React, { FunctionComponent } from "react";
import { View, Text } from "../Themed";
import { Image, StyleSheet } from "react-native";

type SingleResultProps = {
    name: string;
    image: string;
};

const SingleResult: FunctionComponent<SingleResultProps> = ({ name, image }) => {
    return (
        <View style={styles.wrapper}>
            <Image source={{ uri: image }} style={styles.image} />
            <Text style={styles.name}>name</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
    },
    image: {
        width: 312,
        height: 231,
    },
    name: {
        width: 312,
        textAlign: "center",
        lineHeight: 1.5,
    },
});

export default SingleResult;
