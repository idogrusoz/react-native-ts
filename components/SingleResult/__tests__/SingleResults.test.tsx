import { render, RenderAPI } from "@testing-library/react-native";
import React from "react";
import SingleResult from "../SingleResult";
jest.mock("react-redux", () => ({
    useDispatch: jest.fn(),
}));
jest.mock("@react-navigation/native", () => ({
    useNavigation: jest.fn(),
}));

describe("Single result test", () => {
    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(<SingleResult id={1} image="image" title="name" />);
    });
    it("renders witthout crash", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
