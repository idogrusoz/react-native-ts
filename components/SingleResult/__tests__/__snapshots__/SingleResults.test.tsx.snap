// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single result test renders witthout crash 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "image",
          }
        }
        style={
          Object {
            "height": 115,
            "width": 156,
          }
        }
      />
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontFamily": "System",
            },
            Object {
              "textAlign": "center",
              "width": 115,
            },
          ]
        }
      >
        name
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
